src/backend/distributed/operations/worker_shard_copy.c

```diff
@@ -317,6 +389,11 @@ ShardCopyDestReceiverShutdown(DestReceiver *dest)
 
 		PQclear(result);
 		ForgetResults(copyDest->connection);
+
+		/*Reset Replication Origin. */
+		ExecuteCriticalRemoteCommand(copyDest->connection,
```

Comment:
What happens if we move multiple shards to the same node? They would all be using the same replication origin name. Does that cause issues?

Reply:
related, what happens if the replication origin is abandoned? e.g. is there a limit?

Reply:
the limit is set to PG_UINT16_MAX
https://github.com/postgres/postgres/blob/REL_14_STABLE/src/backend/replication/logical/origin.c#L273

Reply:
Added the source node id, to the origin id citus_cdc_<id> so that it will be unique for moving shards from different nodes to a same node.